docs/source/content/supervised/knn.rst

```diff
@@ -2,32 +2,160 @@
 k-Nearest Neighbors
 ====================
 
-K-Nearest Neighbors (KNN) is a basic classifier for machine learning. So we are trying to identify what class an object is in. To do this we look at the closest points (neighbors) to the object and the class with the majority of neighbors will be the class that we identify the object to be in. The k is the number of nearest neighbors to the object. So if k = 1 then the class the object would be in is the class of the closest neighbor. Let's look at an example.
+.. contents::
+  :local:
+  :depth: 3
+
+Introduction
+-------------
+
+K-Nearest Neighbors (KNN) is a basic classifier for machine learning. 
+So, we are trying to identify what class an object is in. To do this we 
+look at the closest points (neighbors) to the object and the class with 
+the majority of neighbors will be the class that we identify the object 
+to be in. The k is the number of nearest neighbors to the object. So, if 
+k = 1 then the class the object would be in is the class of the closest 
+neighbor. Let's look at an example.
 
 .. figure:: _img/knn.png
    :scale: 50 %
    :alt: KNN
 
    Ref: https://coxdocs.org
 
-So in this example we are trying to classify the red star to be either a green square or a blue octagon. So first if we look at the inner circle where k = 3, we can see that there are 2 blue octagons and 1 green square. So there is a majority of blue octagons, so the red star would be classified as a blue octagon. Now we take a look at k = 5, the outer circle. In this one there is 2 blue octagons and 3 green squares. So the red star would be classified as a green square.
+In this example we are trying to classify the red star to be either 
+a green square or a blue octagon. First, if we look at the inner circle
+where k = 3, we can see that there are 2 blue octagons and 1 green square.
+So there is a majority of blue octagons, so the red star would be classified
+as a blue octagon. Now we look at k = 5, the outer circle. In this one
+there is 2 blue octagons and 3 green squares. Then, the red star would be 
+classified as a green square.
 
 How does it work?
 -----------------
 
-We will look at two different ways to go about this. The two ways we will look at is the brute force method and the K-D tree method.
+We will look at two different ways to go about this. The two ways are
+the brute force method and the K-D tree method.
 
 Brute Force Method
 --------------------
 
-This is the simpliest method. Basically it's just calculating the euclidean distance from the object being classified to each point in the set. You want to use this method when the dimensions are small or the number of points are small.
+This is the simplest method. Basically, it's just calculating the Euclidean 
```

Comment:
define Euclidean distance (simply), for those that might not know the term but are probably aware of the concept